### PR TITLE
import count test1

### PR DESCRIPTION
**Pull request by jcommelin**
- **chore: Split `Algebra.GroupPower.Order` Move its lemmas (mostly) to two new files: * `Algebra.Order.Group.Basic` for the lemmas about group-like objects * `Algebra.Order.Ring.Basic` for the lemmas about ring-like objects**
- **ci: report changes to the import graph in a spoiler comment**
- **Apply suggestions from code review**
- **update action, permissions of import-graph-report**
- **Use Mathlib**
- **get improvements and fixes from Damiano**

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: .github/workflows/import_count.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      run: |
+        # the checkout dance, to avoid a detached head
+        git checkout master
+        git checkout -
+
+        # Compute the counts for the HEAD of the PR
+        python ./scripts/count-trans-deps.py "Mathlib/" > head.json
+
+        # Checkout the merge base
+        git checkout master # let's start easy! "$(git merge-base refs/remotes/origin/main ${{ github.sha }})"
```
Comment (adomani) [inline .github/workflows/import_count.yml:43]:
```suggestion
        git checkout "$(git merge-base master ${{ github.sha }}))"
```
According to [stackoverflow](https://stackoverflow.com/questions/20808892/git-diff-between-current-branch-and-master-but-not-including-unmerged-master-com) this is the commit that `...` uses.  (I have not tested that it too much).

File: .github/workflows/import_count.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      run: |
+        message=$(python ./scripts/import-graph-report.py base.json head.json changed_files.txt)
+        echo "${message}"
+        data=$(jq -n --arg msg "$message" '{"body": $msg}')
+        method="POST"
+        if [[ -n "$message" ]]; then
+            url="https://api.github.com/repos/${GITHUB_REPOSITORY}/issues/${{ github.event.pull_request.number }}/comments"
+            headers="Authorization: token ${GITHUB_TOKEN}"
+            comments=$(curl -s -S -H "Content-Type: application/json" -H "$headers" "$url")
+            comment_id=$(echo "$comments" | jq -r '.[] | select(.body | startswith("### Import summary")) | .id')
```
Comment (adomani) [inline .github/workflows/import_count.yml:63]:
```suggestion
            comment_id=$(curl -s -S -H "Content-Type: application/json" -H "$headers" "$url" |
               jq -r '.[] | select(.body | startswith("### Import summary")) | .id')
```

File: .github/workflows/import_count.yml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+      run: |
+        python -m pip install --upgrade pip
+        sudo apt-get install -y jq
+        # If you have additional dependencies, install them here
+
+    - name: Get changed files
+      run: |
+        git fetch origin ${{ github.base_ref }}  # fetch the base branch
+        git diff --name-only origin/${{ github.base_ref }}... > changed_files.txt  # get the list of changed files
+        cat changed_files.txt
```
Comment (adomani) [inline .github/workflows/import_count.yml:31]:
```suggestion
```
Consider removing this.

File: .github/workflows/import_count.yml [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+        # Compute the counts for the merge base
+        python ./scripts/count-trans-deps.py "Mathlib/" > base.json
+
+    - name: Compare results and comment
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: |
+        message=$(python ./scripts/import-graph-report.py base.json head.json changed_files.txt)
+        echo "${message}"
```
Comment (adomani) [inline .github/workflows/import_count.yml:56]:
```suggestion
```
This may also be unnecessary.

File: .github/workflows/import_count.yml (hunk 2 above)
Comment (adomani) [reply .github/workflows/import_count.yml:63]: Also, is `comment_id` actually a line-break-separated list of comment id's that fit the query, or is it just the first one?

**Comment by adomani** [comment]
### Import summary

Just trying to trick your script!  :smile:

**Comment by Ruben-VandeVelde** [comment]
Can we close this and https://github.com/leanprover-community/mathlib4/pull/13564?

**Pull request closed without merging**